src/rust/src/wallet.rs

```diff
@@ -224,8 +233,9 @@ impl Wallet {
     /// in place with the expectation that they will be overwritten and/or updated in
     /// the rescan process.
     pub fn reset(&mut self) {
-        for tx_notes in self.wallet_received_notes.values_mut() {
+        for tx_notes in self.wallet_note_positions.values_mut() {
             tx_notes.tx_height = None;
+            tx_notes.note_positions.clear();
         }
```

Comment:
With the suggestion above, this would just become `self.wallet_note_positions.clear();`.